src/OpenFeatureSDK/OpenFeatureClient.cs

```diff
@@ -241,7 +273,7 @@ public FeatureClient(FeatureProvider featureProvider, string name, string versio
                 flagKey,
                 defaultValue,
                 flagValueType, this._metadata,
-                OpenFeature.Instance.GetProviderMetadata(),
+                provider.GetMetadata(),

```

Comment:
We don't want to call `GetProviderMetadata` here on the `OpenFeature` singleton, because it could have change between when we got the resolution method and here. We want to be sure to only use the cached reference.